lib/dependencies/HarmonyDetectionParserPlugin.js

```diff
@@ -60,6 +70,15 @@ module.exports = class HarmonyDetectionParserPlugin {
 					"Top-level-await is only supported in EcmaScript Modules"
 				);
 			}
+			if (runtimeTemplate.supportsAsyncFunction() === false) {
```

Comment:
maybe add separate plugin for this, since it could be another `async` modules ( wasm, externals ) and better add warning once to compilation